frontend/lib/api/types/recipe.ts

```diff
@@ -206,7 +206,7 @@ export interface Recipe {
   recipeCategory?: RecipeCategory[];
   tags?: RecipeTag[];
   tools?: RecipeTool[];
-  rating?: number;
+  rating?: number | null;
```

Comment:
I don't think you need this because it's already optional. These types are auto-generated from the backend anyway

Reply:
Typescript doesn't like accepting `null` for an optional type:
![image](https://github.com/mealie-recipes/mealie/assets/647846/da50c6c3-cc3d-4029-9816-acd4ac1d277c)

I missed that they're just generated though so I'll go tweak that instead.

Reply:
I haven't been able to get the generation to work--I get this error after running `make generate`:
`Module Error: json2ts must be installed. Instructions can be found here: https://www.npmjs.com/package/json-schema-to-typescript`
Any method I try to install that doesn't resolve it.

Given [this comment](https://github.com/mealie-recipes/mealie/pull/526#issuecomment-862478853), are we _actually_ using the generation method or is it still a little hacky? Even once we get auto-generated types to work I suspect there will be problems explicitly having it generate a type that accepts `null`, so I'm perfectly happy adjusting the type in `recipe.ts` if we aren't actually using it as a generated type.

Reply:
Ah that's funny, I've been using it a bunch. That's fine then, if it's a problem we can fix it later. Not really in the scope of this PR to figure that out anyway. Good work!